utils/utils.go

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package utils
+
+// RemoveArrayDuplicate elements
+func RemoveArrayDuplicate(tagArr []string) []string {
+	// Use map to record duplicates if found
+	encountered := map[string]bool{}
+	result := []string{}
+
+	for element := range tagArr {
+		if encountered[tagArr[element]] == true {
```

Comment:
Perhaps just test for false and remove the empty code block ?   Perhaps it might also be cleaner to use a map and store the elements as the keys which would make them unique, then write the map out as a new array.

Reply:
Reworked this logic to make better use of the map

Reply:
Looks better thanks!